modules/security/src/main/java/io/oasp/module/security/common/api/accesscontrol/AbstractAccessControl.java

```diff
@@ -84,7 +84,7 @@ public boolean equals(Object obj) {
     if (getClass() != obj.getClass()) {
       return false;
     }
-    AccessControl other = (AccessControl) obj;
+    AbstractAccessControl other = (AbstractAccessControl) obj;
```

Comment:
@amarinso , if you see the overridden "equals" method  , it has the snippet the following snippet . If we are casting to interface , 'other.id' will give compilation errors because other is an interface reference . Any suggestions in this case ? 

if (!Objects.equals(this.id, other.id)) {